src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,37 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        self.temp_model_name = None
```

Comment:
nit: what's the reason for `self.` here? let's try and not mutate the customer's model more than we have to.